src/Core/Services/MetadataProviders/ISqlMetadataProvider.cs

```diff
@@ -47,6 +47,11 @@ public interface ISqlMetadataProvider
         /// </summary>
         string? GetSchemaGraphQLFieldTypeFromFieldName(string entityName, string fieldName);
 
+        /// <summary>
+        /// Gets a collection of linking entities generated by DAB (required to support nested mutations).
+        /// </summary>
+        IReadOnlyDictionary<string, Entity> GetLinkingEntities() => new Dictionary<string, Entity>();
```

Comment:
Yes. Instead of throwing a not implemented exception (which kind of breaks substitution principle as well - trying best to follow those), it helps if we just return an empty dictionary. This also helps avoid a lot of conditionals which would need to be introduced. Example: the function `GraphQLSchemaCreator.GenerateObjectDefinitionsForLinkingEntities()` is just being used without any conditional to check that we are dealing with MsSql and not any other db type. Later when we support nested mutation for those db types as well, we won't need to deal with much conditional logic.

https://github.com/Azure/data-api-builder/blob/ff3e952808991c71789c7bd130a97530e21c2c0e/src/Core/Services/GraphQLSchemaCreator.cs#L294